pandas/_libs/groupby.pyx

```diff
@@ -253,18 +253,16 @@ def group_cumsum(numeric[:, ::1] out,
                         t = accum[lab, j] + y
                         compensation[lab, j] = t - accum[lab, j] - y
                         accum[lab, j] = t
-                        out[i, j] = accum[lab, j]
+                        out[i, j] = t
                     else:
                         out[i, j] = NaN
                         if not skipna:
                             accum[lab, j] = NaN
                             break
                 else:
-                    y = val - compensation[lab, j]
-                    t = accum[lab, j] + y
-                    compensation[lab, j] = t - accum[lab, j] - y
+                    t = val + accum[lab, j]
```

Comment:
umm this is affecting all dtypes. do we not have tests for this for small floats?

Reply:
This is inside the else block from an if statement `if numeric == float32_t or numeric == float64_t:` so only non-floats should end up here.

Reply:
Have added a comment